rfcs/72-editor.md

```diff
@@ -0,0 +1,508 @@
+   function named `cli`,  this will be the main entry point for the CLI and
+   graphical editor.
+3. Add a `main.rs` file to the `src/` folder associated with the `bevy` crate.
+   Call `cli` from that crate.
+    1. This makes two `bevy` crates, a library one, which already exists, and an
+       executable one, which will be what the user installs.
+4. Add `bevy_editor_command` as a dependency for `bevy`, this can be done
+   through the `bevy_internal` crate, or as an optional feature.
+5. The `cli` function must do basic argument parsing, project templating, and
+   system/credential management. It will dispatch everything else to the `bevy_editor` crate.
+
+
+### Version dispatching
+
+1. Any project using the `bevy` tool is expected to have an `editor` binary
+   crate in their workspace, the template will fill this in automatically.
+2. The dispatch can use `cargo`, as a simple example, `bevy editor` can become
+   an alias for:
+```
+cargo run -p editor
+```
+3. For this reason, the `editor` crate must provide the main function, so the
+   dispatch should ideally have graceful failing if it doesn't exist.
+4. We are using a separate crate so that the end distributable does not need to
+   rely on the editor, developers that _do_ want the deliverable to rely on the editor can call it themselves.
+
+### CLI Interface
+
+1. Simple command line argument parsing can be done with clap.
+2. Simple error handling can be provided with anyhow.
+3. There does not seem to be a standard pretty-output library for rust,
+   prettytable seems to be a start.
+   1. Options like crossterm might also fit the bill, but also seem like overkill.
+4. The manipulator/live viewing components can delegate to the window interface.
+
+### Window Interface
+
+1. Extend from the base crate's world.
+1. Ideally, everything should be do-able through the graphical interface, but
+   some things can be pushed to later releases.
+2. The `bevy_editor_command` window only needs to handle: (1)Credential Storage,
+   (2) Template Downloads, and (3) Project Selection, everything else is handled
+   by the a separate `bevy_editor` window.
+3. The `bevy_editor_command` window must seamlessly transition to the
+   `bevy_editor` window, as if opening a project.
+4. The ui should be made using bevy components.
+
+### New project construction
+
+1. Handled by `bevy_editor_command`.
+2. Retrieve template from asset repo or local template storage.
+3. Build template object mapping template languages to their result.
+    1. For example foo.rs.tera should become foo.rs.
+4. Fill in appropriate template fields from the user's environment.
+5. Add CI workflows.
+6. Add licenses.
+7. Initialize as Git repo.
+
+### Asset management
+
+1. Template management can be done using terra.
+2. Downloading static assets can be managed by reqwest.
+3. Downloading code assets can be done with cargo.
+4. Downloading template assets can be done with libgit2.
```

Comment:
```suggestion
4. Downloading template assets can be done with `libgit2`.
```

Reply:
Maybe [gitoxide](https://github.com/Byron/gitoxide) is a better fit for this, one less wrapped C library. Makes builds more predictable and stable on different platforms.